facades/PC/build.gradle.kts

```diff
@@ -30,396 +28,370 @@ dateTimeFormat.timeZone = TimeZone.getTimeZone("UTC")
  *
  * @return
  */
-def String nativeSubdirectoryName() {
-    if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-        return "windows"
-    } else if (Os.isFamily(Os.FAMILY_MAC)) {
-        return "macosx"
-    } else if (Os.isFamily(Os.FAMILY_UNIX)) {
-        return "linux"
-    } else {
-        logger.warn("What kind of libraries do you use on this? {}", System.properties["os.name"])
-        return "UNKNOWN"
+fun nativeSubdirectoryName(): String {
+    return when {
+        Os.isFamily(Os.FAMILY_WINDOWS) -> "windows"
+        Os.isFamily(Os.FAMILY_MAC) -> "macosx"
+        Os.isFamily(Os.FAMILY_UNIX) -> "linux"
+        else -> {
+            logger.warn("What kind of libraries do you use on this? {}", System.getProperty("os.name"))
+            "UNKNOWN"
+        }
     }
 }
 
 
-ext {
-    // Default path to store server data if running headless via Gradle
-    localServerDataPath = 'terasology-server'
+// Default path to store server data if running headless via Gradle
+val localServerDataPath by extra("terasology-server")
 
-    // General props
-    mainClassName = 'org.terasology.engine.Terasology'
-    subDirLibs = 'libs'
-    templatesDir = new File(rootDir, 'templates')
-    rootDirDist = new File(rootDir, 'build/distributions')
+// General props
+val mainClassName by extra("org.terasology.engine.Terasology")
+val subDirLibs = "libs"
+val templatesDir = File(rootDir, "templates")
+val rootDirDist = File(rootDir, "build/distributions")
 
-    // Read environment variables, including variables passed by jenkins continuous integration server
-    env = System.getenv()
+// Inherited props
+val dirNatives: String by rootProject.extra
+val distsDirectory: DirectoryProperty by project;
+
+// Read environment variables, including variables passed by jenkins continuous integration server
+val env: MutableMap<String, String> = System.getenv()!!
+
+// Version related
+val startDateTimeString = dateTimeFormat.format(Date())!!
+val versionFileName = "VERSION"
+val versionBase by lazy { File(templatesDir, "version.txt").readText().trim() }
+val displayVersion = versionBase
 
-    // Version related
-    startDateTimeString = dateTimeFormat.format(new Date())
-    versionFileName = 'VERSION'
-    versionBase = new File(templatesDir, "version.txt").text.trim()
-    displayVersion = versionBase
-}
```

Comment:
This isn't strictly line-for-line compatible because I only kept _some_ of these as gradle "extra" properties. Most of them I left as local variables.

I don't believe it's a problem for the normal build environment, since there's nothing else in the build that interacts with them, but maybe it could be a concern if something like Jenkins has an alternate configuration that sets or uses them differently?